packages/aws-cdk-lib/aws-iam/lib/instance-profile.ts

```diff
@@ -0,0 +1,183 @@
+import { Construct } from 'constructs';
+import { CfnInstanceProfile } from './iam.generated';
+import { IRole } from './role';
+import { Resource, Arn, Stack, IResource } from '../../core';
+
+/**
+ * Represents an IAM Instance Profile
+ */
+export interface IInstanceProfile extends IResource {
+  /**
+   * The InstanceProfile's name.
+   * @attribute
+   */
+  readonly instanceProfileName: string;
+
+  /**
+   * The InstanceProfile's ARN.
+   * @attribute
+   */
+  readonly instanceProfileArn: string;
+
+  /**
+   * The role associated with the InstanceProfile.
+   *
+   * @default - no role
+   */
+  readonly role?: IRole;
+}
+
+/**
+ * Properties of an Instance Profile
+ */
+export interface InstanceProfileProps {
+  /**
+   * The name of the role to associate with the InstanceProfile. Only one role can
+   * be assigned to an EC2 instance at a time, and all applications on the instance
+   * share the same role and permissions.
+   */
+  readonly role: IRole;
```

Comment:
I keep going back and forth on this, but I'm leaning towards making this optional and creating a role for you by default.

Reply:
This makes sense. I'll get this updated.